### PR TITLE
Enable Refreshing and Store extensions

### PR DESCRIPTION
**Pull request by alari**

### CODE REVIEW

File: kademlia/src/main/scala/fluence/kad/conf/AdvertizeConf.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fluence.kad.conf
+
+case class AdvertizeConf(host: String, port: Short)
```
Comment (DieMyst) [inline kademlia/src/main/scala/fluence/kad/conf/AdvertizeConf.scala]: docs, what is `advertize`?

File: kademlia/src/main/scala/fluence/kad/conf/JoinConf.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fluence.kad.conf
+
+case class JoinConf(seeds: Seq[String], numOfNodes: Int)
```
Comment (DieMyst) [inline kademlia/src/main/scala/fluence/kad/conf/JoinConf.scala]: docs

File: kademlia/src/main/scala/fluence/kad/conf/KademliaConfig.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fluence.kad.conf
+
+case class KademliaConfig(
```
Comment (DieMyst) [inline kademlia/src/main/scala/fluence/kad/conf/KademliaConfig.scala]: docs

File: kademlia/src/main/scala/fluence/kad/contact/UriContact.scala [hunk 4]
```diff
@@ -148,28 +149,29 @@ object UriContact {
...
 
-    val readHost: Uri ~~> String = (uri: Uri) ⇒
-      Either.fromOption(uri.host, CodecError("Host not provided")).map(_.value)
+    val readHost: URI ~~> String = (uri: URI) ⇒ Either.fromOption(Option(uri.getHost), CodecError("Host not provided"))
 
-    val readPort: Uri ~~> Short = (uri: Uri) ⇒
+    val readPort: URI ~~> Short = (uri: URI) ⇒
       Either
-        .fromOption(uri.port, CodecError("Port not provided"))
+        .fromOption(Option(uri.getPort), CodecError("Port not provided"))
```
Comment (DieMyst) [inline kademlia/src/main/scala/fluence/kad/contact/UriContact.scala]: `getPort` couldn't be `null`, but could be `-1` if undefined

Comment (alari) [reply kademlia/src/main/scala/fluence/kad/contact/UriContact.scala]: Thanks!

**Pull request merged**